src/libusermode/userhook.hpp

```diff
@@ -113,6 +115,12 @@
 #include "plugins/plugins_ex.h"
 #include "printers/printers.hpp"
 
+class UserHookException : public std::runtime_error
```

Comment:
I'm not a fan of custom exceptions - it's very "python like" and not "C++ like", but I guess it doesn't hurt.

Reply:
IMHO the least amount of crazy C++ is used the better :) Just because it allows you to do it doesn't mean you should. In light of @BonusPlay 's comment, is there any other more standard way to get what you need done?

Reply:
Sure, I'll use `std::runtime_error` directly.